client/sockettransport.go

```diff
@@ -79,7 +79,7 @@ func (c *socketTransport) call(handler server.Handler, request interface{}, resp
 
 	// read response
 	var (
-		responseBytes  = []byte{}
+		responseBytes  = make([]byte, 0, 8192)
```

Comment:
Whats the reason to setting the maximum buffer size to 8192 bytes?

Please implement a unit test to show that this is robust in all edge cases.

Reply:
The 8192 is just a wild guess how large the incoming data can be. Usually you should measure it and then you can set the pre-allocated buffer. I will try to add a test.

Reply:
This function is untestable in a unit test. requires to much refactoring. But you should guess yourself that pre-allocating a byte slice with a appropriate capacity avoids rebuilding the internal slice when calling `append`.